File: software/examples/python/example_simple.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-  

HOST = "localhost"
PORT = 4223
UID = "XYZ" # Change to your UID

from tinkerforge.ip_connection import IPConnection
from tinkerforge.bricklet_hall_effect import HallEffect

if __name__ == "__main__":
    ipcon = IPConnection() # Create IP connection
    he = HallEffect(UID, ipcon) # Create device object

    ipcon.connect(HOST, PORT) # Connect to brickd
    # Don't use device before ipcon is connected

    # Get current edge count of encoder without reset 
    edge_count = he.get_edge_count(False)

    print('EdgeCount: ' + str(edge_count))

    raw_input('Press key to exit\n') # Use input() in Python 3
    ipcon.disconnect()

File: software/examples/javascript/ExampleSimple.js
var IPConnection = require('Tinkerforge/IPConnection');
var BrickletHallEffect = require('Tinkerforge/BrickletHallEffect');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'irP';// Change to your UID

var ipcon = new IPConnection();// Create IP connection
var he = new BrickletHallEffect(UID, ipcon);// Create device object

ipcon.connect(HOST, PORT,
    function(error) {
        console.log('Error: '+error);        
    }
);// Connect to brickd

// Don't use device before ipcon is connected
ipcon.on(IPConnection.CALLBACK_CONNECTED,
    function(connectReason) {
        // Get current edge count of encoder without reset 
        he.getEdgeCount(false,
            function(edgeCount) {
                console.log('Edge Count: '+edgeCount);
            },
            function(error) {
                console.log('Error: '+error);
            }
        );
    }
);

console.log("Press any key to exit ...");
process.stdin.on('data',
    function(data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

